### PR TITLE
Implement more PE-D and other enhancements

### LINKED ISSUE #105: [PE-D] faculty and email under contact_manager
Should the format of email be enforced such that only the format of email can be accepted? i.e xxxxxx@xxxx.xxx.xxxx, xxxxxx@xxxx.xxx etc.

Is it better to only allow certain Faculty input? i.e SOC, FOE etc.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: a1021492980/ped#3

### PR DESCRIPTION
**Pull request by wli-linda**
Resolves #105

### CODE REVIEW

File: src/main/java/seedu/allonus/contacts/ContactsManager.java [hunk 1]
```diff
@@ -63,10 +64,17 @@
...
+            "*** Note: You didn't update this name / only updated casing! ***";
+    private static final String CONTACTS_SET_DUPLICATE_NAME_MESSAGE =
+            "A contact with the same name already exists!";
+
     private static final Logger logger = Logger.getLogger("");
     private static final int CONTACTS_LIST_MAX_SIZE = 100;
     private static final ArrayList<Contact> listOfContacts =
             new ArrayList<>(CONTACTS_LIST_MAX_SIZE);
+    private static final HashSet<Integer> nameHashes =
+            new HashSet<>(CONTACTS_LIST_MAX_SIZE);
```
Comment (YitHien) [inline src/main/java/seedu/allonus/contacts/ContactsManager.java:77]: Interesting way to check for uniqueness, looks good!

**Review by YitHien — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/allonus/contacts/ContactsManager.java [hunk 2]
```diff
@@ -95,55 +139,51 @@ private static void contactsWelcome() {
...
 
-    private static void listContacts() {
+    public static void checkUniqueName(String newName, String oldName) throws InvalidContactField {
+        String lowerOldName = oldName.toLowerCase();
+        String lowerNewName = newName.toLowerCase();
+        int oldNameHash = lowerOldName.hashCode();
+        int newNameHash = lowerNewName.hashCode();
+
+        boolean existsInList = nameHashes.contains(newNameHash);
+        boolean isCaseChange = (oldNameHash == newNameHash);
```
Comment (mathanmahe) [inline src/main/java/seedu/allonus/contacts/ContactsManager.java:149]: Good use of hash for checking existing names!

**Review by mathanmahe — APPROVED** [review]
LGTM!

**Pull request merged**